gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GitFlowGraphMonitor.java

```diff
@@ -0,0 +1,332 @@
+
+    String repositoryUri = config.getString(ConfigurationKeys.GIT_FLOWGRAPH_MONITOR_REPO_URI);
+
+    this.repositoryDir = ConfigUtils.getString(config, ConfigurationKeys.GIT_FLOWGRAPH_MONITOR_REPO_DIR,
+        ConfigurationKeys.DEFAULT_GIT_FLOWGRAPH_MONITOR_REPO_DIR);
+
+    String branchName = ConfigUtils.getString(config, ConfigurationKeys.GIT_FLOWGRAPH_MONITOR_BRANCH_NAME,
+        ConfigurationKeys.DEFAULT_GIT_FLOWGRAPH_MONITOR_BRANCH_NAME);
+
+    Integer pollingInterval = ConfigUtils.getInt(config, ConfigurationKeys.GIT_FLOWGRAPH_MONITOR_POLLING_INTERVAL,
+        ConfigurationKeys.DEFAULT_GIT_FLOWGRAPH_MONITOR_POLLING_INTERVAL);
+
+    this.flowGraphDir = ConfigUtils.getString(config, ConfigurationKeys.GIT_FLOWGRAPH_MONITOR_FLOWGRAPH_DIR,
+        ConfigurationKeys.DEFAULT_GIT_FLOWGRAPH_MONITOR_FLOWGRAPH_DIR);
+
+    this.flowGraphDirPath = new Path(this.repositoryDir, this.flowGraphDir);
+    this.javaPropsExtensions = Sets.newHashSet(Splitter.on(",").split(PROPERTIES_EXTENSIONS));
+    this.hoconFileExtensions = Sets.newHashSet(Splitter.on(",").split(CONF_EXTENSIONS));
+
+    try {
+      this.pullFileLoader = new PullFileLoader(this.flowGraphDirPath,
+          FileSystem.get(URI.create(ConfigurationKeys.LOCAL_FS_URI), new Configuration()),
+          this.javaPropsExtensions, this.hoconFileExtensions);
+    } catch (IOException e) {
+      throw new RuntimeException("Could not create pull file loader", e);
+    }
+
+    initRepo(repositoryUri, repositoryDir, branchName, pollingInterval);
+  }
+
+  /**
+   * Determine if the service should poll Git.
+   */
+  @Override
+  public boolean shouldPollGit() {
+    return this.isActive;
+  }
+
+  /**
+   * Add an element (i.e., a {@link DataNode}, or a {@link FlowEdge} to
+   * the {@link FlowGraph} for an added, updated or modified node or edge file.
+   * @param change
+   */
+  @Override
+  public void addChange(DiffEntry change) {
+    Path path = new Path(change.getNewPath());
+    if(path.depth() == 3) {
+      addDataNode(change);
+    } else if(path.depth() == 4) {
+      addFlowEdge(change);
+    }
+  }
+
+  /**
+   * Remove an element (i.e. either a {@link DataNode} or a {@link FlowEdge} from the {@link FlowGraph} for
+   * a renamed or deleted {@link DataNode} or {@link FlowEdge} file.
+   * @param change
+   */
+  @Override
+  public void removeChange(DiffEntry change) {
+    Path path = new Path(change.getOldPath());
+    if(path.depth() == 3) {
+      removeDataNode(change);
+    } else if(path.depth() == 4) {
+      removeFlowEdge(change);
+    }
+  }
+
+  /**
+   * Add a {@link DataNode} to the {@link FlowGraph}. The method uses the {@link FlowGraphConfigurationKeys#DATA_NODE_CLASS} config
+   * to instantiate a {@link DataNode} from the node config file.
+   * @param change
+   */
+  private void addDataNode(DiffEntry change) {
+    if(checkFilePath(change.getNewPath(), NODE_FILE_DEPTH)) {
```

Comment:
Please add space after the `if`s in the code. Checkstyle checks for this.

Reply:
Added space after ifs.